packages/lab/src/multiline-input/MultilineInput.tsx

```diff
@@ -0,0 +1,194 @@
+import { clsx } from "clsx";
+import {
+  ChangeEvent,
+  ComponentPropsWithoutRef,
+  FocusEvent,
+  forwardRef,
+  ReactNode,
+  Ref,
+  TextareaHTMLAttributes,
+  useState,
+} from "react";
+import { makePrefixer, StatusAdornment, useControlled } from "@salt-ds/core";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+
+import multilineInputCss from "./MultilineInput.css";
+
+const withBaseName = makePrefixer("saltMultilineInput");
+
+export interface MultilineInputProps
+  extends Omit<ComponentPropsWithoutRef<"div">, "defaultValue">,
+    Pick<
+      ComponentPropsWithoutRef<"input">,
+      "disabled" | "value" | "defaultValue" | "placeholder"
+    > {
+  /**
+   * End adornment component
+   */
+  endAdornment?: ReactNode;
+  /**
+   * Styling variant with full border. Defaults to false
+   */
+  fullBorder?: boolean;
+  /**
+   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#Attributes) applied to the `textarea` element.
+   */
+  textAreaProps?: TextareaHTMLAttributes<HTMLTextAreaElement>;
+  /**
+   * Optional ref for the textarea component
+   */
+  textAreaRef?: Ref<HTMLTextAreaElement>;
+  /**
+   * If `true`, the component is read only.
+   */
+  readOnly?: boolean;
+  /**
+   * Start adornment component
+   */
+  startAdornment?: ReactNode;
+  /**
+   * Validation status.
+   */
+  validationStatus?: "error" | "warning" | "success";
```

Comment:
there seem to be some inconsistencies on the `validationStatus` type. Maybe we can re-think this when adding the new statuses